Builds/CMake/CMakeFuncs.cmake

```diff
@@ -147,11 +147,7 @@ macro(setup_build_cache)
     set(CMAKE_CONFIGURATION_TYPES
       DebugClassic
       ReleaseClassic)
-    if(${CMAKE_BUILD_TYPE} STREQUAL "Debug")
-      set(CMAKE_BUILD_TYPE DebugClassic)
-    elseif(${CMAKE_BUILD_TYPE} STREQUAL "Release")
-      set(CMAKE_BUILD_TYPE ReleaseClassic)
-    endif()
+    set(CMAKE_BUILD_TYPE "${CMAKE_BUILD_TYPE}Classic")
```

Comment:
There is a bug here. If the build type is already `ReleaseClassic` this makes it `ReleaseClassicClassic` and the flags are wrong. To repro, create a build with `cmake -Dtarget=gcc.release.nounity`